### ISSUE TITLE
RestrictionValidator.validatePattern: invalid matching+ typo

### ISSUE THREAD

**Issue by d6c8** [xmlet/XsdAsmFaster#2]
This method is defined as:
```java
public static void validatePattern(String pattern, String string){
        if (!string.replaceAll(pattern, "").equals(string)){
            throw new RestrictionViolationException("Violation of pattern restriction, the string doesn't math the acceptable pattern, which is " + pattern);
        }
    }
```

It fails in the following scenario:
```java
    // ...
    link().attrSizes("16x16").__()
   // ...
```

1. It should probably be:
```java
 if (!string.replaceAll(pattern, "").equals("")){
```
There are probably more efficient solutions to check matching, such as:
```java
 if (!Pattern.matches(pattern, string)){
```

2. _doesn't math_ should be _doesn't match_

**Comment by lcduarte** [comment]
Hello,

Thanks for using the library and for the suggestion! 

Added your fixes and deployed the 1.0.7 version.

**Comment by d6c8** [comment]
Thanks, I'm currently experimenting different solutions to generate HTML and will keep you informed.
